### PR TITLE
[matchers] Add Username matcher.

### PR DESCRIPTION
**Pull request by acs**
Implemented this matcher sto cover the use case in which we have usernames from GitHub and we use them to unify identities.

### CODE REVIEW

File: tests/test_matcher_username.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                                 Identity(name='Smith. J', source='mls'),
+                                 Identity(name='Smith. J', email='JSmith@example.com', source='mls')]
+
+        jsmith_alt = UniqueIdentity(uuid='J. Smith')
+        jsmith_alt.identities = [Identity(name='J. Smith', username='john_smith', source='alt'),
+                                 Identity(name='John Smith', username='jsmith', source='alt'),
+                                 Identity(email='', source='alt'),
+                                 Identity(email='jsmith', source='alt')]
+
+        jsmith_not_email = UniqueIdentity(uuid='John Smith')
```
Comment (sduenas) [inline tests/test_matcher_username.py]: These identities are not used here. Remove them.

File: tests/test_matcher_username.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        matcher = UsernameMatcher()
+
+        # First two unique does not produce any match
+        result = matcher.match(jsmith, john_smith)
+        self.assertEqual(result, False)
+
+        result = matcher.match(john_smith, jsmith)
+        self.assertEqual(result, False)
+
+        # Comparing with the third match with the second
```
Comment (sduenas) [inline tests/test_matcher_username.py]: This is a wrong comment.

File: tests/test_matcher_username.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+        result = matcher.match_filtered_identities(john_alt, jsmith_uuid)
+        self.assertEqual(result, True)
+
+        # Although the UUID is equal to None, these two does not match
+        result = matcher.match_filtered_identities(jsmith_none, jdoe_none)
+        self.assertEqual(result, False)
+
+    def test_match_filtered_identities_instances(self):
+        """Test whether it raises an error when ids are not EmailNameIdentities"""
```
Comment (sduenas) [inline tests/test_matcher_username.py]: Fix this comment.

File: sortinghat/matching/username.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        if not isinstance(fb, UsernameIdentity):
+            raise ValueError("<fb> is not an instance of UsernameIdentity")
+
+        if fa.uuid and fb.uuid and fa.uuid == fb.uuid:
+            return True
+
+        if fa.username in self.blacklist:
+            return False
+
+        # Compare username addresses first
```
Comment (sduenas) [inline sortinghat/matching/username.py]: Fix this comment too. We're not talking about addresses

**Comment by sduenas** [comment]
Reviewed. Please fix the problems and squash the commits you may need.

File: tests/test_matcher_username.py (hunk 1 above)
Comment (acs) [inline tests/test_matcher_username.py]: Done! Ups, sorry about that!

File: tests/test_matcher_username.py (hunk 2 above)
Comment (acs) [inline tests/test_matcher_username.py]: Fixed!

File: tests/test_matcher_username.py (hunk 3 above)
Comment (acs) [inline tests/test_matcher_username.py]: Fixed. I would promise that I changed in all document EmailNameIdentities with UsernameIdentities ... I don't know how this one survived! ;)

File: sortinghat/matching/username.py (hunk 4 above)
Comment (acs) [inline sortinghat/matching/username.py]: Done!

**Comment by acs** [comment]
Ok, all comments addressed. Time to to the squashing and push a new PR!

**Comment by acs** [comment]
PR ready! Could you review it @sduenas?

**Pull request merged**